### PR TITLE
fix for comparing manual add to autocomplete

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#14175

*Description of changes:*
- Fix for when a user manually enters the same secured party as one added from the autocomplete search/dropdown
- The address objects from the search do not have the optional fields, so this fix removes the optional fields (if empty) from the manually added party to allow them to be compared.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
LGTM! 
The cleaning method looks like it could be a generic utility method, for reuse where required.
If you feel like refactoring later to make it more global, don't hesitate to do so :)

File: ppr-ui/src/components/parties/composables/useSecuredParty.ts [hunk 1]
```diff
@@ -95,6 +95,15 @@ export const useSecuredParty = (props, context) => {
     return idx !== -1
   }
 
+  const removeEmptyAttributes = (address: AddressIF): AddressIF => {
+    // removes any undefined/empty attributes from party clone so it can be compared to autocomplete objects
+    const keys = Object.keys(address)
+    keys.forEach(key => {
+      if (!address[key]) delete address[key]
```
Comment (dimak1) [inline ppr-ui/src/components/parties/composables/useSecuredParty.ts]: Is the idea here to mutate the address object? Is it intentional?

**Comment by owillborn** [comment]
> 

Oooo good idea!

File: ppr-ui/src/components/parties/composables/useSecuredParty.ts (hunk 1 above)
Comment (owillborn) [reply ppr-ui/src/components/parties/composables/useSecuredParty.ts]: Yes I want to change the clone of the address object. The addresses from the autocomplete search don't have the optional fields, but they still need to be compared for equality in a duplicate Secured Party. So removing any undefined/empty attributes allows them to be compared properly

**Comment by dimak1** [comment]
@owillborn you could also explore [lodash](https://lodash.com/) methods, maybe you can find something to help with the removal of empty values.

**Comment by owillborn** [comment]
> @owillborn you could also explore [lodash](https://lodash.com/) methods, maybe you can find something to help with the removal of empty values.

Ill take a look, thanks!

File: ppr-ui/src/components/parties/composables/useSecuredParty.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/components/parties/composables/useSecuredParty.ts]: ok thanks for clarification

**Review by dimak1 — APPROVED** [review]

**Comment by owillborn** [comment]
> @owillborn you could also explore [lodash](https://lodash.com/) methods, maybe you can find something to help with the removal of empty values.

@dimak1 found something much nicer, thanks!

File: ppr-ui/package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "core-js": "^3.6.5",
     "flush-promises": "^1.0.2",
     "http-status-codes": "^2.1.4",
+    "lodash-clean": "^2.2.3",
```
Comment (dimak1) [inline ppr-ui/package.json:32]: hmm not sure about importing a library for just one function

**Comment by owillborn** [comment]
@dimak1 @cameron-eyds Commit history got really messy, gonna close this and create a new one, sorry

**Comment by dimak1** [comment]
@owillborn just in case, you should be doing a **rebase** not **merge** when syncing with `main` branch. I'll send you a link.

**Pull request closed without merging**